onnx/backend/test/case/node/negativeloglikelihoodloss.py

```diff
@@ -40,12 +55,19 @@ def compute_negative_log_likelihood_loss(input, target, weight=None, reduction='
                 for i in range(input_shape[0]):
                     if target[i] == ignore_index:
                         gather_weight[i] = 0
-
-            if len(input_shape) == 3:
+            elif len(input_shape) == 3:
                 for i in range(input_shape[0]):
-                    for j in range(input_shape[1]):
+                    for j in range(input_shape[2]):
                         if target[i][j] == ignore_index:
                             gather_weight[i][j] = 0
+            elif len(input_shape) == 4:
```

Comment:
humm... if you have case for `len(input_shape) == 3`, why do you need another case? We can flatten `[N, C, H, W]` to `[N, C, H*W]` and apply the 3-D branch.